### PR TITLE
Error Fixes: debugging null pointer exceptions and other errors

### LINKED ISSUE #81: [Error fixes 1] Null Pointer Exception Fixes and Making the Code Run

### PR DESCRIPTION
**Pull request by e-laflamme**
# Description

PR includes changes by Harsh (@harshsbajwa), Yassin (@BetterthanNothin) and Emily (@e-laflamme). Focused on resolving exceptions/errors and getting the program to run with all intended features. 

Updates include:
- Main UI can be refreshed, now UI will show user changes like task creation and adopting a pet. 
- User can now use task list function without a pet. 
- User can now customized their pet and unlock customizations
- Roulette button will give error message when task list is empty
- Deadlines are automatically set for the end of the day (potential for future feature to allow user to select time)
- Added tests for TaskView UI
- Updates tests so that they do not rely on a specific deadline being past or present
- User can now add and modify tasks
- User will receive a popup warning if they entered task information incorrectly
- Completed tasks will display with no checkbox or edit button

Fixes #81  (issue)

## Type of change

- [x] Compilation error fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation
- [x] Test cases
- [ ] Redundant code removal (fix code smells)
- [ ] Data storage 
- [x] UI element

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by XinyingXu — APPROVED** [review]
This is fantastic, you have done such an amazing job. Our UI looks so adorable, and seeing everything working feels like a dream come true!

**Review by harshsbajwa — APPROVED** [review]
looks good, refer to discord discussion for potential changes

File: src/main/java/entities/User.java [hunk 1]
```diff
@@ -23,6 +26,8 @@ public User (int points) {
     public User (int points, Pet pet) {
         this.points = points;
```
Comment (stephaniesaab) [inline src/main/java/entities/User.java:27]: This can maybe call the empty constructor as a helper and then reset the points.

**Review by stephaniesaab — APPROVED** [review]
Code looks good and functional! Some refactoring is needed and there are some points of redundancy. Those can be better addressed in a later PR. Changes to the tests and use cases improved the integration by a lot.

**Pull request merged**